CLI UDF rename command

scholtzan opened the pull request:
Fixes https://github.com/mozilla/bigquery-etl/issues/1341

This `bqetl udf rename` and `bqetl mozfun rename` command supports renaming UDFs and UDF datasets:

```
# to rename datasets:
bqetl mozfun rename bits28 -n new_bits28

# to rename UDFs:
bqetl mozfun rename bits28.range -n bits28.better_range
bqetl mozfun rename bits28.range -n new_bits28.better_range
bqetl mozfun rename bits28.range -n new_bits28.range
```

fbertsch commented on bigquery_etl/cli/udf.py:
```diff
@@ -266,3 +269,74 @@ def publish(ctx, path, project, dependency_dir, gcs_bucket, gcs_path):
+        match = UDF_NAME_RE.match(new_name)
+        new_udf_name = match.group("name")
+        new_udf_dataset = match.group("dataset")
+    elif UDF_DATASET_RE.match(name) and UDF_DATASET_RE.match(new_name):
```
Should we instead use globbing to indicate renaming a dataset? `bqetl rename from-dataset.* to-dataset`. This may be simpler to reason about since it's similar to `mv`

fbertsch reviewed (commented):
Just a few thoughts on usability. Other than that this is super cool and _very_ useful.

scholtzan commented:
Updated the command to allow wildcard patterns and removed the `-n` option. So, now it can be used like:

```
bqetl mozfun rename bits28.range new_bits28.range
bqetl mozfun rename bits28.* new_bits28
bqetl mozfun rename bits28.range new_bits28  # rename to new_bits28.range
```

fbertsch reviewed (approved):
Oh yeah, this is very exciting. The new patterns is familiar.
